paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java

```diff
@@ -364,6 +364,31 @@ default void repairTable(Identifier identifier) throws TableNotExistException {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Exception for trying to operate on a resource that doesn't have permission. Define as a
+     * runtime exception: 1. Other engine has no this type exception. 2. It wouldn't bring api break
```

Comment:
Just keep "Exception for trying to operate on a resource that doesn't have permission."

Reply:
It is better to separate database and table? And expose database like DatabaseNotEmptyException, and expose identifier like TableAlreadyExistException.